src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import core.basesyntax.operations.OperationHandler;
+import core.basesyntax.operations.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<String, Integer> storage = new HashMap<>();
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(storage, transaction);
+        }
+    }
+
+    @Override
+    public Map<String, Integer> getStorage() {
```

Comment:
The method `getStorage` violates the encapsulation principle by returning a reference to the mutable `storage` object. Instead, return a copy of the storage to prevent external modifications.